plugins/packages/restapi/lib/index.ts

```diff
@@ -169,9 +183,34 @@ export default class RestapiQueryService implements QueryService {
         ...paramsFromUrl,
         ...this.searchParams(sourceOptions, queryOptions, hasDataSource),
       },
-      ...(isUrlEncoded ? { form: json } : { json }),
     };
 
+    const hasFiles = Object.values(json || {}).some((item) => {
+      return isFileObject(item);
+    });
+
+    if (isUrlEncoded) {
+      requestOptions.form = json;
+    } else if (hasFiles) {
+      const form = new FormData();
+      for (const key in json) {
+        if (isFileObject(json[key])) {
+          const fileBuffer = Buffer.from(json[key].base64Data, 'base64');
+          form.append(key, fileBuffer, {
+            filename: json[key].name,
+            contentType: json[key].type,
+            knownLength: fileBuffer.length,
+          });
+        } else {
```

Comment:
Made it so that missing values will result in empty strings.

Reply:
Can you please check [here](https://github.com/sindresorhus/got/issues/674#issuecomment-448213059) and confirm once? because AFAIK, this line 205 will still cause issue if json[key] is undefined.

Reply:
@8chili-sanju tried with `undefined` and `null` values. Both caused an issue with `form-data`, so I added a check to guard against those values and ignore it.